playbooks/cleanup.yaml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Cleanup
+  hosts: all
+  roles:
+  - role: ensure-pip  # https://zuul-ci.org/docs/zuul-jobs/latest/python-roles.html#role-ensure-pip
+  vars:
+    cloud_provider: "{{ cloud }}"  # inherited from the parent job
+    prefix: "pr{{ zuul.change }}"
+    testcluster_name: "pr{{ zuul.change }}"
+  environment:
+    OS_CLOUD: "{{ cloud_provider }}"
+    PATH: "{{ ansible_user_dir }}/.local/bin:{{ ansible_env.PATH }}"
+  tasks:
+  - name: Ensure ospurge and openstackclient
+    pip:
+      name:
+      - "git+https://git.openstack.org/openstack/ospurge"
+      - "python-openstackclient"
+      extra_args: --user
+  - name: Cleanup - keypair
+    command: "openstack keypair delete {{ prefix }}-keypair"
```

Comment:
Use the openstack ansible collection: https://docs.ansible.com/ansible/latest/collections/openstack/cloud/index.html

Reply:
Good idea, this will addressed within https://github.com/SovereignCloudStack/k8s-cluster-api-provider/issues/568